src/contracts/core/AllocationManagerStorage.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/utils/structs/DoubleEndedQueue.sol";
+
+import "../interfaces/IAllocationManager.sol";
+import "../interfaces/IDelegationManager.sol";
+
+import {Snapshots} from "../libraries/Snapshots.sol";
+
+abstract contract AllocationManagerStorage is IAllocationManager {
+    using Snapshots for Snapshots.DefaultWadHistory;
+
+    // Constants
+
+    /// @dev Index for flag that pauses operator allocations/deallocations when set.
+    uint8 internal constant PAUSED_MODIFY_ALLOCATIONS = 0;
+
+    /// @dev Index for flag that pauses operator register/deregister to operator sets when set.
+    uint8 internal constant PAUSED_OPERATOR_SLASHING = 1;
+
+    /// @dev Index for flag that pauses operator register/deregister to operator sets when set.
+    uint8 internal constant PAUSED_OPERATOR_SET_REGISTRATION_AND_DEREGISTRATION = 2;
+
+    // Immutables
+
+    /// @notice The DelegationManager contract for EigenLayer
+    IDelegationManager public immutable delegation;
+
+    /// @notice Delay before deallocations are clearable and can be added back into freeMagnitude
+    /// In this window, deallocations still remain slashable by the operatorSet they were allocated to.
+    uint32 public immutable DEALLOCATION_DELAY;
+
+    /// @notice Delay before alloaction delay modifications take effect.
```

Comment:
typo: `Delay before alloaction delay modifications take effect.` => `Delay before allocation modifications take effect.`

also there's some 'edge case' behavior here, right? I recall this not being enforced on deallocations if the Operator isn't slashable by the AVS? can we be any more specific in this definition to reflect this?